app/services/wallet_transactions/validate_service.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module WalletTransactions
+  class ValidateService
+    def initialize(result, **args)
+      @result = result
+      @args = args
+    end
+
+    def valid?
+      errors = []
+      errors << valid_wallet?
+      errors << valid_paid_credits_amount?
+      errors << valid_granted_credits_amount?
+      errors = errors.compact
+
+      unless errors.empty?
+        result.fail!(
+          code: 'unprocessable_entity',
+          message: 'Validation error on the record',
+          details: errors,
+        )
+        return false
+      end
+
+      true
+    end
+
+    private
+
+    attr_accessor :result, :args
+
+    def valid_wallet?
+      current_wallet = Wallet.find_by(
```

Comment:
Can we scope this to `current_organization` maybe?

`current_organization.wallets.find_by(...)`